panels/info/cc-info-panel.c

```diff
@@ -484,117 +479,59 @@ get_os_description (void)
-  g_unix_mount_free (mount);
-
-  g_file_query_filesystem_info_async (file,
-                                      G_FILE_ATTRIBUTE_FILESYSTEM_SIZE,
-                                      0,
-                                      self->priv->cancellable,
-                                      (GAsyncReadyCallback) query_done,
-                                      self);
-  g_object_unref (file);
-}
-
-static void
-get_primary_disc_info (CcInfoPanel *self)
-{
-  GList        *points;
-  GList        *p;
-  GHashTable *hash;
+  manager = udisks_client_get_object_manager (self->priv->client);
+  objects = g_dbus_object_manager_get_objects (manager);
+  added_drives = g_hash_table_new (g_str_hash, g_str_equal);
 
-  hash = g_hash_table_new (g_str_hash, g_str_equal);
-  points = g_unix_mount_points_get (NULL);
+  for (l = objects; l != NULL; l = l->next)
+    {
+      UDisksDrive *drive;
 
-  /* If we do not have /etc/fstab around, try /etc/mtab */
-  if (points == NULL)
-    points = g_unix_mounts_get (NULL);
+      drive = udisks_object_peek_drive (UDISKS_OBJECT (l->data));
 
-  for (p = points; p != NULL; p = p->next)
-    {
-      GUnixMountEntry *mount = p->data;
-      const char *mount_path;
-      const char *device_path;
-
-      mount_path = g_unix_mount_get_mount_path (mount);
-      device_path = g_unix_mount_get_device_path (mount);
-
-      /* Do not count multiple mounts with same device_path, because it is
-       * probably something like btrfs subvolume. Use only the first one in
-       * order to count the real size. */
-      if (gsd_should_ignore_unix_mount (mount) ||
-          gsd_is_removable_mount (mount) ||
-          g_str_has_prefix (mount_path, "/media/") ||
-          g_str_has_prefix (mount_path, g_get_home_dir ()) ||
-          g_hash_table_lookup (hash, device_path) != NULL)
+      /* Skip removable devices */
+      if (!drive ||
+          udisks_drive_get_removable (drive) ||
+          udisks_drive_get_ejectable (drive) ||
+          g_hash_table_contains (added_drives, udisks_drive_get_id (drive)))
         {
-          g_unix_mount_free (mount);
           continue;
         }
 
-      self->priv->primary_mounts = g_list_prepend (self->priv->primary_mounts, mount);
-      g_hash_table_insert (hash, (gpointer) device_path, (gpointer) device_path);
+      total_size += udisks_drive_get_size (drive);
+      g_hash_table_add (added_drives, (gpointer) udisks_drive_get_id (drive));
     }
-  g_list_free (points);
-  g_hash_table_destroy (hash);
 
-  self->priv->cancellable = g_cancellable_new ();
-  get_primary_disc_info_start (self);
+  if (total_size > 0)
+    {
+      g_autofree gchar *size = g_format_size (total_size);
+      gtk_label_set_text (GTK_LABEL (label), size);
+    }
+  else
+    {
+      gtk_label_set_text (GTK_LABEL (label), "-");
+    }
+
+  g_list_free_full (objects, g_object_unref);
+  g_clear_pointer (&added_drives, g_hash_table_destroy);
+  g_clear_object (&manager);
```

Comment:
Nitpick: You can use `g_autoptr()` on the declarations to avoid having to do this memory management manually. (g-c-c depends on GLib 2.44, which is when `g_autoptr()` was introduced). What’s here is correct though.

Note: When I say ‘nitpick’ in a review, that means it’s something which is not a blocker for the PR being accepted. Fix it if you want, or justify not fixing it or deferring a fix — either is fine.